patches/tModLoader/Terraria/WorldGen.cs.patch

```diff
@@ -929,68 +929,58 @@
  		Main.tile[num6 + 1, num + 2].frameY = (short)(num8 + 36);
  		Main.tile[num6 + 1, num + 2].frameX = (short)(num2 + 18);
  		Main.tile[num6 + 1, num + 2].UseBlockColors(cache3);
-@@ -27008,11 +_,15 @@
+@@ -27008,10 +_,12 @@
  			return;
  
  		destroyObject = true;
-+		// Extra context.
++		bool drop = TileLoader.Drop(x, j, type);
  		for (int k = 0; k < num; k++) {
  			if (Main.tile[x, num5 + k].type == type)
  				KillTile(x, num5 + k);
  		}
++		using var _disableNewItem = new Item.DisableNewItemMethod(!drop);
```

Comment:
Is there a specific reason these should be separate? It'd be nice to have single line patches. 

I'm thinking of the use-case where people want to replace the vanilla drops for a tile with something else. Is this common? 

I suppose the right way to do that would be to run their own drop code in `CanDrop`? 
But `CanDrop` is meant to be 'pure'. If modders actually did drop items there, they wouldn't be compatible with another mod attempting to block drops. 
We often make `Can` hook short-circuit with this in mind.

If overriding vanilla drops is meant to be supported, we need a little thought (and maybe an example).

Reply:
Yes, KillTile might cause other tiles to drop. For example, mining a stack of Bars with a Table placed on top. NewItem needs to be disabled after KillTile to not mistakenly disable drops from other tiles that might be dropping from the KillTile->FrameTile recursion. 

We do not currently support replacing drops. To do that, I'd think the existing drop would need to be collected, but we aren't doing that currently.